nixos/doc/manual/development/reviewing-contributions.xml

```diff
@@ -0,0 +1,286 @@
+
+<para>Pull-requests reviews should include a list of what has been reviewed in a comment, so other reviewers and mergers can know the state of the review.</para>
+
+<para>All the review template samples provided in this section are generic and meant as examples. Their usage is optional and the reviewer is free to adapt them to his liking.</para>
+
+<section><title>Package updates</title>
+
+<para>A package update is the most trivial and common type of pull-request. These pull-requests mainly consist in updating the version part of the package name and the source hash.</para>
+<para>It can happen that non trivial updates include patches or more complex changes.</para>
+
+<para>Reviewing process:</para>
+
+<itemizedlist>
+  <listitem><para>Add labels to the pull-request. (Requires commit rights)</para>
+    <itemizedlist>
+      <listitem><para><literal>8.has: package (update)</literal> and any topic label that fit the updated package.</para></listitem>
+    </itemizedlist>
+  </listitem>
+  <listitem><para>Ensure that the package versioning is fitting the guidelines.</para></listitem>
+  <listitem><para>Ensure that the commit text is fitting the guidelines.</para></listitem>
+  <listitem><para>Ensure that the package maintainers are notified.</para>
+    <itemizedlist>
+      <listitem><para>mention-bot usually notify github users based on the submitted changes, but it can happen that it misses some of the package maintainers.</para></listitem>
+    </itemizedlist>
+  </listitem>
+  <listitem><para>Ensure that the meta field contains correct information.</para>
+    <itemizedlist>
+      <listitem><para>License can change with version updates, so it should be checked to be fitting upstream license.</para></listitem>
+      <listitem><para>If the package has no maintainer, a maintainer must be set. This can be the update submitter or a community member that accepts to take maintainership of the package.</para></listitem>
+    </itemizedlist>
+  </listitem>
+  <listitem><para>Ensure that the code contains no typos.</para></listitem>
+  <listitem><para>Building the package locally.</para>
+    <itemizedlist>
+      <listitem><para>Pull-requests are often targeted to the master or staging branch so building the pull-request locally as it is submitted can trigger a large amount of source builds.</para>
+        <para>It is possible to rebase the changes on nixos-unstable or nixpkgs-unstable for easier review by running the following commands from a nixpkgs clone.
+<screen>
+$ git remote add channels https://github.com/NixOS/nixpkgs-channels.git <co xml:id='reviewing-rebase-1' />
+$ git fetch channels nixos-unstable <co xml:id='reviewing-rebase-2' />
+$ git fetch origin pull/PRNUMBER/head:PRNUMBER <co xml:id='reviewing-rebase-3' />
```

Comment:
I usually just prefix the local branch name with `pr-`: `git fetch origin pull/PRNUMBER/head:pr-PRNUMBER` to avoid any potential clash and still have it named in a convenient way.